src/Documents.php

```diff
@@ -62,41 +62,83 @@ private function endPointPath(string $action = ''): string
      */
     public function create(array $document): array
     {
-        return $this->apiCall->post($this->endPointPath(''), $document);
+        return $this->apiCall->post($this->endPointPath(''), $document, true);
+    }
+
+    /**
+     * @param array $document
+     *
+     * @return array
+     * @throws TypesenseClientError|GuzzleException
+     */
+    public function upsert(array $document): array
+    {
+        return $this->apiCall->post($this->endPointPath(''), $document, true, ['action' => 'upsert']);
+    }
+
+    /**
+     * @param array $document
+     *
+     * @return array
+     * @throws TypesenseClientError|GuzzleException
+     */
+    public function update(array $document): array
+    {
+        return $this->apiCall->post($this->endPointPath(''), $document, true, ['action' => 'update']);
     }
 
     /**
      * @param array $documents
+     * @param array $options
      *
      * @return array
      * @throws TypesenseClientError|GuzzleException|\JsonException
      */
-    public function createMany(array $documents): array
+    public function createMany(array $documents, array $options = []): array
     {
-        $res = $this->import(
-            implode(
-                "\n",
-                array_map(
-                    static fn(array $document) => json_encode($document, JSON_THROW_ON_ERROR),
-                    $documents
-                )
-            )
+        $this->apiCall->getLogger()->warning(
+            "createMany is deprecated and will be removed in a future version. " .
+            "Use import instead, which now takes both an array of documents or a JSONL string of documents"
         );
-        return array_map(static function ($item) {
-            return json_decode($item, true, 512, JSON_THROW_ON_ERROR);
-        }, explode("\n", $res));
+        return $this->import($documents, $options);
     }
 
     /**
-     * @param string $documents
+     * @param string|array $documents
+     * @param array $options
      *
-     * @return string
+     * @return string|array
      * @throws TypesenseClientError
      * @throws GuzzleException
+     * @throws \JsonException
      */
-    public function import(string $documents): string
+    public function import($documents, array $options = [])
     {
```

Comment:
$options added here, needs to be referenced in the documentation of the client.

Reply:
Yup, it's documented here: 

https://github.com/typesense/typesense-website/blob/1aa626f37491fa41dac969bfbc34dc705405d30f/typesense.org/docs/0.16.0/api.html#L1246-L1254